lib/net/ssh/buffer.rb

```diff
@@ -262,27 +261,28 @@ def read_private_keyblob(type)
           _unkown2 = read_bignum
           dmp1 = d % (p - 1)
           dmq1 = d % (q - 1)
-          if key.respond_to?(:set_key)
-            key.set_key(n, e, d)
-          else
-            key.e = e
-            key.n = n
-            key.d = d
-          end
-          if key.respond_to?(:set_factors)
-            key.set_factors(p, q)
-          else
-            key.p = p
-            key.q = q
+          # Public key
+          data_sequence = OpenSSL::ASN1::Sequence([
+                                                    OpenSSL::ASN1::Integer(n),
+                                                    OpenSSL::ASN1::Integer(e),
+                                                  ])
+
+          if d && p && q && dmp1 && dmq1 && iqmp
+            data_sequence = OpenSSL::ASN1::Sequence([
```

Comment:
Since we're in the `read_private_keyblob` method, wouldn't it make more sense to just abort/raise an exception rather than going through the motions of actually creating the pubkey?

Reply:
I'm definitely out of my depth, but why would we want to raise an error here?

The existing code constructs a `PKey` but uses the newly deprecated `set_key`, `set_factors` and `set_crt_params` methods when doing so.  I was under the impression that building a key directly from the ASN.1 gets us the same result, but has the benefit of not using the deprecated methods.